### PR TITLE
[5.5] fix for broken queue connections in worker

### PR DESCRIPTION
**Pull request by martianoff**
If queue connection fails queue workers produce errors and never restarts
Ideally we should think how to move Failed connection handler to third parties connection interface or Job class. So third party developers or job task developers can extend it

### CODE REVIEW

File: src/Illuminate/Queue/Worker.php [hunk 1]
```diff
@@ -5,14 +5,17 @@
...
 use Illuminate\Contracts\Debug\ExceptionHandler;
 use Symfony\Component\Debug\Exception\FatalThrowableError;
 use Illuminate\Contracts\Cache\Repository as CacheContract;
 
 class Worker
 {
-    use DetectsLostConnections;
+    use \Illuminate\Database\DetectsLostConnections, DetectsLostConnections {
+        \Illuminate\Database\DetectsLostConnections::causedByLostConnection insteadof DetectsLostConnections;
+        \Illuminate\Database\DetectsLostConnections::causedByLostConnection as causedByLostDbConnection;
```
Comment (fernandobandeira) [inline src/Illuminate/Queue/Worker.php]: I think it would be better to add an use statement for this with an alias instead of calling it everytime, would make it a little bit cleaner.

**Comment by taylorotwell** [comment]
Will have to be sent to 5.5.

**Comment by taylorotwell** [comment]
The use statement is crazy. Can we just make the trait extend the other trait somehow?

**Comment by taylorotwell** [comment]
Close pending inactivity. I don't want that crazy use statement. Hard to follow.

**Comment by milsanore** [comment]
Also having this issue in a distributed Laravel architecture. One RabbitMQ network fault disables that node from doing any subsequent work until supervisord is restarted. Thx
